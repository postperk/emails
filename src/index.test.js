import * as emails from './index';
import orderMockData from './testData/order.json';
import brandMockData from './testData/brand.json';
import fs from 'fs-extra';

console.info = jest.fn();

describe('Emails', () => {
   it('Offer email will match snapshot', async () => {
      const compiled = await emails.compile('offer', {
         brand: brandMockData,
         order: orderMockData
      });

      await fs.writeFile('./generated/offer.html', compiled.content);
      expect(compiled.content).toMatchSnapshot();
   });
});
