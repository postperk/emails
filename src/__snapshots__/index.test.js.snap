// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emails Offer email will match snapshot 1`] = `

<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:v="urn:schemas-microsoft-com:vml"
      xmlns:o="urn:schemas-microsoft-com:office:office"
>
  <head>
    <title>
      UGG Royale
    </title>
    <!--[if !mso]><!-- -->
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta http-equiv="Content-Type"
          content="text/html; charset=UTF-8"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <style type="text/css">
      #outlook a { padding:0; }
          .ReadMsgBody { width:100%; }
          .ExternalClass { width:100%; }
          .ExternalClass * { line-height:100%; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
    </style>
    <!--[if !mso]><!-->
    <style type="text/css">
      @media only screen and (max-width:480px) {
            @-ms-viewport { width:320px; }
            @viewport { width:320px; }
          }
    </style>
    <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
    <!--[if lte mso 11]>
        <style type="text/css">
          .outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
    <!--[if !mso]><!-->
    <link href="http://fonts.googleapis.com/css?family=Open+Sans"
          rel="stylesheet"
          type="text/css"
    >
    <link href="https://fonts.googleapis.com/css?family=Sorts+Mill+Goudy"
          rel="stylesheet"
          type="text/css"
    >
    <style type="text/css">
      @import url(http://fonts.googleapis.com/css?family=Open+Sans);
@import url(https://fonts.googleapis.com/css?family=Sorts+Mill+Goudy);
    </style>
    <style type="text/css">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
.mj-column-px-500 { width:500px !important; max-width: 500px; }
.mj-column-px-300 { width:300px !important; max-width: 300px; }
      }
    </style>
    <style type="text/css">
      @media only screen and (max-width:480px) {
      table.full-width-mobile { width: 100% !important; }
      td.full-width-mobile { width: auto !important; }
    }
    </style>
    <style type="text/css">
      @media all and (max-width: 600px) {
        img[title*="img30"] {
          width: 30% !important;
        }
      }
    </style>
  </head>
  <body style="background-color:#ffffff;">
    <div style="background-color:#ffffff;">
      <!--[if mso | IE]>
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
      <div style="background:#ffffff;background-color:#ffffff;Margin:0px auto;max-width:600px;">
        <table align="center"
               border="0"
               cellpadding="0"
               cellspacing="0"
               role="presentation"
               style="background:#ffffff;background-color:#ffffff;width:100%;"
        >
          <tbody>
            <tr>
              <td style="direction:ltr;font-size:0px;padding:0px;text-align:center;vertical-align:top;">
                <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
            <tr>
              <td
                 class="" width="600px"
              >
          
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
                <div style="Margin:0px auto;max-width:600px;">
                  <table align="center"
                         border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="width:100%;"
                  >
                    <tbody>
                      <tr>
                        <td style="direction:ltr;font-size:0px;padding:0px;text-align:center;vertical-align:top;">
                          <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:600px;"
            >
          <![endif]-->
                          <div class="mj-column-per-100 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:0px;word-break:break-word;"
                                >
                                  <table border="0"
                                         cellpadding="0"
                                         cellspacing="0"
                                         role="presentation"
                                         style="border-collapse:collapse;border-spacing:0px;"
                                  >
                                    <tbody>
                                      <tr>
                                        <td style="width:170px;">
                                          <a href="http://www.karmacandleclub.com/"
                                             target="_blank"
                                          >
                                            <img alt="Karma Candle Club"
                                                 height="auto"
                                                 src="https://firebasestorage.googleapis.com/v0/b/pperk-dashboard-dev.appspot.com/o/karma%20logo.png?alt=media&token=57f3b527-a109-418d-9c85-8ef7e8d50d01"
                                                 style="border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;"
                                                 width="170"
                                            >
                                          </a>
                                        </td>
                                      </tr>
                                    </tbody>
                                  </table>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
              </td>
            </tr>
          
            <tr>
              <td
                 class="" width="600px"
              >
          
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
                <div style="Margin:0px auto;max-width:600px;">
                  <table align="center"
                         border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="width:100%;"
                  >
                    <tbody>
                      <tr>
                        <td style="direction:ltr;font-size:0px;padding:10px 0 20px;text-align:center;vertical-align:top;">
                          <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:600px;"
            >
          <![endif]-->
                          <div class="mj-column-per-100 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Sorts Mill Goudy, Garamond, Times, serif;font-size:30px;line-height:36px;text-align:center;color:#111111;">
                                    Louis, here's an eGift card on us!
                                  </div>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
              </td>
            </tr>
          
            <tr>
              <td
                 class="" width="600px"
              >
          
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
                <div style="Margin:0px auto;max-width:600px;">
                  <table align="center"
                         border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="width:100%;"
                  >
                    <tbody>
                      <tr>
                        <td style="direction:ltr;font-size:0px;padding:0px;text-align:center;vertical-align:top;">
                          <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:500px;"
            >
          <![endif]-->
                          <div class="mj-column-px-500 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:16px;line-height:26px;text-align:center;color:#555555;">
                                    Thanks for your order. We've joined with these brand partners that we think you'll love. Please enjoy the following on us:
                                  </div>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
              </td>
            </tr>
          
            <tr>
              <td
                 class="" width="600px"
              >
          
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
                <div style="background:#ffffff;background-color:#ffffff;Margin:0px auto;max-width:600px;">
                  <table align="center"
                         border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="background:#ffffff;background-color:#ffffff;width:100%;"
                  >
                    <tbody>
                      <tr>
                        <td style="direction:ltr;font-size:0px;padding:20px 0 0 0;text-align:center;vertical-align:top;">
                          <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:300px;"
            >
          <![endif]-->
                          <div class="mj-column-px-300 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:0px;word-break:break-word;"
                                >
                                  <table border="0"
                                         cellpadding="0"
                                         cellspacing="0"
                                         role="presentation"
                                         style="border-collapse:collapse;border-spacing:0px;"
                                  >
                                    <tbody>
                                      <tr>
                                        <td style="width:300px;">
                                          <a href="http://www.ugg.com/"
                                             target="_blank"
                                          >
                                            <img alt="Guy on a beach with an UGGxGrain surfboard"
                                                 height="auto"
                                                 src="https://firebasestorage.googleapis.com/v0/b/pperk-dashboard-dev.appspot.com/o/card_allgirls.png?alt=media&token=58bd759b-e9cd-4034-b2e6-4ac1cd6aa508"
                                                 style="border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;"
                                                 width="300"
                                            >
                                          </a>
                                        </td>
                                      </tr>
                                    </tbody>
                                  </table>
                                </td>
                              </tr>
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:12px;line-height:18px;text-align:center;color:#555555;">
                                    <p class="tagline"
                                       style="min-height: 36px; margin: 0; padding: 0;"
                                    >
                                      Premium razors delivered to you monthly
                                    </p>
                                  </div>
                                </td>
                              </tr>
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:12px;line-height:18px;text-align:center;color:#555555;">
                                    Gift card code:
                                    <br>
                                    <a href="https://www.google.com"
                                       class="code-link"
                                       style="font-weight: bold; font-size: 15px; text-decoration: none; color: #555;"
                                    >
                                      4354-WKM-AH300
                                    </a>
                                  </div>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
            <td
               class="" style="vertical-align:top;width:300px;"
            >
          <![endif]-->
                          <div class="mj-column-px-300 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:0px;word-break:break-word;"
                                >
                                  <table border="0"
                                         cellpadding="0"
                                         cellspacing="0"
                                         role="presentation"
                                         style="border-collapse:collapse;border-spacing:0px;"
                                  >
                                    <tbody>
                                      <tr>
                                        <td style="width:300px;">
                                          <a href="http://www.ugg.com/"
                                             target="_blank"
                                          >
                                            <img alt="Guy on a beach with an UGGxGrain surfboard"
                                                 height="auto"
                                                 src="https://firebasestorage.googleapis.com/v0/b/pperk-dashboard-dev.appspot.com/o/card_happy.png?alt=media&token=7f1fb59d-bbc0-4c5d-a289-25f8e3b6187f"
                                                 style="border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;"
                                                 width="300"
                                            >
                                          </a>
                                        </td>
                                      </tr>
                                    </tbody>
                                  </table>
                                </td>
                              </tr>
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:12px;line-height:18px;text-align:center;color:#555555;">
                                    <p class="tagline"
                                       style="min-height: 36px; margin: 0; padding: 0;"
                                    >
                                      Premium razors delivered to you monthly
                                    </p>
                                  </div>
                                </td>
                              </tr>
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:10px 25px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:12px;line-height:18px;text-align:center;color:#555555;">
                                    Gift card code:
                                    <br>
                                    <a href="https://www.google.com"
                                       class="code-link"
                                       style="font-weight: bold; font-size: 15px; text-decoration: none; color: #555;"
                                    >
                                      0274-WKM-AH300
                                    </a>
                                  </div>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
              </td>
            </tr>
          
            <tr>
              <td
                 class="" width="600px"
              >
          
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
                <div style="Margin:0px auto;max-width:600px;">
                  <table align="center"
                         border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="width:100%;"
                  >
                    <tbody>
                      <tr>
                        <td style="direction:ltr;font-size:0px;padding:0px;text-align:center;vertical-align:top;">
                          <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:600px;"
            >
          <![endif]-->
                          <div class="mj-column-per-100 outlook-group-fix"
                               style="font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                          >
                            <table border="0"
                                   cellpadding="0"
                                   cellspacing="0"
                                   role="presentation"
                                   style="vertical-align:top;"
                                   width="100%"
                            >
                              <tr>
                                <td align="center"
                                    style="font-size:0px;padding:0 10px;word-break:break-word;"
                                >
                                  <div style="font-family:Open Sans, Arial, sans-serif;font-size:10px;line-height:14px;text-align:center;color:#111111;">
                                    <p class="footer-tag"
                                       style="margin: 25px 0 1ppx 0; padding: 10px 0 0 0; font-size: 10px; line-height: 15px; color: #999;"
                                    >
                                      You're receiving this email in response to your order with Karma Candle Club
                                      <br>
                                      You will not receive further emails
                                    </p>
                                  </div>
                                </td>
                              </tr>
                            </table>
                          </div>
                          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
              </td>
            </tr>
          
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    </div>
  </body>
</html>

`;
